test/parallel/test-runner-snapshot-tests.js

```diff
@@ -0,0 +1,312 @@
+    const manager = new SnapshotManager(__filename, false);
+    const output = manager.serialize({ foo: 1 }, [
+      (value) => { return JSON.stringify(value); },
+      (value) => { return value + '424242'; },
+    ]);
+
+    t.assert.strictEqual(output, '\n{"foo":1}424242\n');
+  });
+
+  test('serialized values get cast to string', (t) => {
+    const manager = new SnapshotManager(__filename, false);
+    const output = manager.serialize(5, []);
+
+    t.assert.strictEqual(output, '\n5\n');
+  });
+
+  test('serialized values get escaped', (t) => {
+    const manager = new SnapshotManager(__filename, false);
+    const output = manager.serialize('fo\\o`${x}`', []);
+
+    t.assert.strictEqual(output, '\nfo\\\\o\\`\\${x}\\`\n');
+  });
+
+  test('reads individual snapshots from snapshot file', (t) => {
+    const fixture = fixtures.path('test-runner', 'snapshots', 'simple.js');
+    const manager = new SnapshotManager(fixture, false);
+    manager.resolveSnapshotFile();
+    manager.readSnapshotFile();
+    const snapshot = manager.getSnapshot('foo 1');
+
+    t.assert.strictEqual(snapshot, '\n{\n  "bar": 1,\n  "baz": 2\n}\n');
+  });
+
+  test('snapshot file is not read in update mode', (t) => {
+    const fixture = fixtures.path('test-runner', 'snapshots', 'simple.js');
+    const manager = new SnapshotManager(fixture, true);
+    manager.readSnapshotFile();
+
+    t.assert.throws(() => {
+      manager.getSnapshot('foo 1');
+    }, /Snapshot 'foo 1' not found/);
+  });
+
+  test('throws if requested snapshot does not exist in file', (t) => {
+    const fixture = fixtures.path('test-runner', 'snapshots', 'simple.js');
+    const manager = new SnapshotManager(fixture, false);
+
+    t.assert.throws(() => {
+      manager.getSnapshot('does not exist 1');
+    }, (err) => {
+      t.assert.strictEqual(err.code, 'ERR_INVALID_STATE');
+      t.assert.match(err.message, /Snapshot 'does not exist 1' not found/);
+      t.assert.strictEqual(err.snapshot, 'does not exist 1');
+      t.assert.strictEqual(err.filename, manager.snapshotFile);
+      return true;
+    });
+  });
+
+  test('snapshot IDs are escaped when stored', (t) => {
+    const fixture = fixtures.path('test-runner', 'snapshots', 'simple.js');
+    const manager = new SnapshotManager(fixture, false);
+
+    manager.setSnapshot('foo`${x}` 1', 'test');
+    t.assert.strictEqual(manager.getSnapshot('foo\\`\\${x}\\` 1'), 'test');
+  });
+
+  test('throws if snapshot file cannot be resolved', (t) => {
+    const manager = new SnapshotManager(null, false);
+    const assertion = manager.createAssert();
+
+    t.assert.throws(() => {
+      assertion('foo');
+    }, (err) => {
+      t.assert.strictEqual(err.code, 'ERR_INVALID_STATE');
+      t.assert.match(err.message, /Invalid snapshot filename/);
+      t.assert.strictEqual(err.filename, null);
+      return true;
+    });
+  });
+
+  test('writes the specified snapshot file', (t) => {
```

Comment:
Add a test for when it fails to write the snapshot file (e.g. becaause of permissions?)